PlaceRouteHierFlow/placer/ILP_solver.cpp

```diff
@@ -465,7 +473,15 @@ double ILP_solver::GenerateValidSolution(design& mydesign, SeqPair& curr_sp, PnR
     }
     HPWL += (HPWL_max_y - HPWL_min_y) + (HPWL_max_x - HPWL_min_x);  
   }
+
   //HPWL norm
+  //
+  // block_HPWL is also not constant across different placements,
+  //    because, again, the placer is choosing different aspect ratios
+  //    of the primitives (which can have different areas.)
+  //    This is the source of the non-constant (and even
+  //    non-monotonic) scaling for hpwl.
```

Comment:
@srini229 This is the source of non-constant hpwl scaling.

Reply:
@stevenmburns thank you. just noticed that as well